### DISCUSSION TITLE
Middleware execute after render page.

### DISCUSSION THREAD

**Discussion by SulivanNext** [nuxt/nuxt#27452] [Questions] [answered]
Hello dears! I have a problem when I disable SSR, which is that the middleware runs after rendering. The screen flashes quickly on the dashboard and immediately after checking whether the session is valid or not, it takes you to the login page. This creates a certain amount of discomfort. Is there any way to resolve this with SSR disabled? 
Thank you in advance

**Comment by chibx** [comment]
@SulivanNext  Can u show us the code for your Nuxt route middleware. If possible, it could be an issue with async/await

**Reply by SulivanNext** [reply]
I'm afraid not, but just in case, here it is!

It is not the best of codes and is under development. ignore things that may not influence, I will refactor it soon. thanks!



`export default defineNuxtRouteMiddleware(async (to, from) => {

    const { error, data }: any = await useAsyncData(() => useServer(endpoints.auth.check));

    if (
        to.path !== '/auth/login' && to.path !== '/auth/forgot-password'
        && to.path !== '/auth/success-mail-send'
        && to.path !== '/auth/reset-password/' + from.params.token
        && !!error.value
    ) {
        return navigateTo('/auth/login');
    }


    if (!error.value) {
        const user = useState('user.me');
        user.value = data.value.user;

    }

    if (to.path == '/auth/login' && !error.value) {
        return navigateTo('/atendimentos');
    }
})`

**Comment by chibx** [comment, marked as answer]
When you disable SSR, the page first shows blank, then loads the JS, and then paints the screen with  your code. 
Before it paints, it runs the middleware assigned to the page, or global ones, then awaits the promises, looking for a redirect/navigate, if there is.
So my idea, is either because of the disabled SSR, or the suspicious useState declared in the middleware. Apart from these, your middleware kinda looks okay, just make sure the conditional statements are okay

**Reply by SulivanNext** [reply]
It works exactly as you say. I discovered that it wasn't a problem with the middleware but with the dynamic layout in root App.vue. Not is the best praticie. Has been fixed. Thanks for the clarification!

**Reply by SulivanNext** [reply]
Solved with setPageLayout('layoutName') composable
